contracts/bonding/BondingManager.sol

```diff
@@ -799,6 +848,8 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
         // Set last round that transcoder called reward
         t.lastRewardRound = currentRound;
 
+        checkpointBondingState(msg.sender, delegators[msg.sender], t);
```

Comment:
Since `updateTranscoderWithRewards()` calls `increaseTotalStake()` which calls `checkpointBondingState()` for the msg.sender in this function, we have two internal calls to `checkpointBondingState()` for the msg.sender in this function - once before `t.lastRewardRound` is updated to the current round and once after.

If L849 is moved into `updateTranscoderWithRewards()` right before the `increaseTotalStake()` call (L1424) I think it would be possible to remove this extra call to `checkpointBondingState()`.

Reply:
Fixed on https://github.com/livepeer/protocol/pull/614/commits/0c41f78401a97ba4d92def55ba21f2c1820ab6b2

On the first implementation I didn't think that this would be a strict requirement, but as I progressed on it and especially after learning about the events interfaces it made a lot more sense to make sure we checkpoint only once per transaction.

The thinking of the original implementation was "whenever the storage is mutated, make sure we checkpoint afterwards" which resulted in the duplicate checkpoints mentioned. I changed this rationale to start from the high-level transaction instead, to make sure that they either checkpoint the state in the end themselves or call some other function (but only once) that checkpoints for them instead.
- Creating a modifier for the checkpointing process made it easier as well so it shows up in the signature of functions that do checkpoint and we can reason about it more quickly.
- There was the need of only 1 function to have an alternate version that doesn't make checkpointing, which I added with the `Uncheckpointed` suffix
- Good thing is I had tests for all those calls already and could verify that all of them were fixed, with only 1 checkpoint per address per tx now